geanyctags/src/geanyctags.c

```diff
@@ -205,6 +205,26 @@ static gchar *generate_find_string(GeanyProject *prj)
 }
 
 
+static const gchar *get_base_path(void)
```

Comment:
this function has a leaky semantic: it returns *either* `prj->base_path` *or* an allocated buffer.  The caller must then free conditionally, which is tricky at best.